ethereum/contracts/Messages.sol

```diff
@@ -26,9 +26,45 @@ contract Messages is Getters {
     *  - it aims to verify the signatures provided against the guardianSet
     */
     function verifyVM(Structs.VM memory vm) public view returns (bool valid, string memory reason) {
+        (valid, reason) = verifyVMInternal(vm, true);    
+    }
+
+    /**
+    * @dev `verifyVMInternal` serves to validate an arbitrary vm against a valid Guardian set
+    *  - it is similar to verifyVM but drops the verification of the hash as it is called on the result of parseVM whose vm.hash field can be trusted.
+    *  - it aims to make sure the VM is for a known guardianSet
+    *  - it aims to ensure the guardianSet is not expired
+    *  - it aims to ensure the VM has reached quorum
+    *  - it aims to verify the signatures provided against the guardianSet
+    */
+    function verifyVMInternal(Structs.VM memory vm, bool checkHash) internal view returns (bool valid, string memory reason) {
         /// @dev Obtain the current guardianSet for the guardianSetIndex provided
         Structs.GuardianSet memory guardianSet = getGuardianSet(vm.guardianSetIndex);
 
+        /**
+         * @dev Verify that the hash field in the vm matches with the hash of the contents of the vm if checkHash is set
```

Comment:
I think natspec comments aren't interpreted in function bodies, so we can drop the "@dev" annotation here